src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SortContext.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.search;
+
+import com.yelp.nrtsearch.server.grpc.QuerySortField;
+import com.yelp.nrtsearch.server.grpc.SearchResponse.Hit.CompositeFieldValue;
+import com.yelp.nrtsearch.server.grpc.SortType;
+import com.yelp.nrtsearch.server.luceneserver.SearchHandler;
+import com.yelp.nrtsearch.server.luceneserver.field.FieldDef;
+import com.yelp.nrtsearch.server.luceneserver.field.properties.HasSortableValueParser;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.lucene.search.FieldDoc;
+import org.apache.lucene.search.Sort;
+import org.apache.lucene.search.SortField;
+
+/**
+ * * Context class to convey all the sorted information, and provides helpers to retrieve the sorted
+ * values.
+ */
+public class SortContext {
```

Comment:
I might suggest restructuring this a bit. The logic could be moved out of the `SortContext` and into `SortParser`, the context would just be a container for processed values.

The `SortParser` could expose something like:
public static Sort parseSort(List<SortType> fields, Map<String, FieldDef> queryFields);
public static SortContext parseSortContext(List<SortType> fields, Map<String, FieldDef> queryFields);
public static Map<String, CompositeFieldValue> getAllSortedValues(FieldDoc fd, SortContext context);

Reply:
fixed